src/pages/home/modals/JoinSessionModal.svelte

```diff
@@ -5,32 +5,35 @@
     import type { GetJamData } from '../../../models/jam';
     import type { AxiosError } from 'axios';
     import { navigate } from 'svelte-navigator';
-    import { JamStore } from '../../../store/jam';
     import { Failure } from '../../../lib/notify/notify';
+    import fuzzysort from 'fuzzysort';
 
     export let closeFunc: Function;
     let jams: GetJamData[];
+    let searchResult: Fuzzysort.KeyResults<GetJamData>;
+    let jamsProgress: string = 'Loading...';
+    let search: string;
 
     function joinJam(id: string) {
-        const jam = jams.find(jam => jam.id === id)
-        JamStore.set({
-            id: jam.id,
-            name: jam.name,
-            bpm: jam.bpm,
-            users: [],
-            ws: null,
-        })
-
-        navigate("/jam", {replace: true})
+        navigate(`/jam/${id}`, { replace: true });
     }
 
     function loadJams() {
-        api.get<{rooms: GetJamData[]}>('/jam').then(({ data }) => {
-            jams = data.rooms;
-        })
-        .catch((error: AxiosError) => {
-            Failure(error.message)
-        })
+        api.get<{ rooms: GetJamData[] }>('/jam')
```

Comment:
We probably want to return this Promise as well
```suggestion
        return api.get<{ rooms: GetJamData[] }>('/jam')
```